src/vs/workbench/api/browser/mainThreadTerminalShellIntegration.ts

```diff
@@ -114,6 +114,7 @@ export class MainThreadTerminalShellIntegration extends Disposable implements Ma
 
 	private _enableShellIntegration(instance: ITerminalInstance): void {
 		this._extensionService.activateByEvent('onTerminalShellIntegration:*');
+		this._register(instance.onDidChangeShellType(() => this._extensionService.activateByEvent(`onTerminalShellIntegration:${instance.shellType}`)));
```

Comment:
Same as https://github.com/microsoft/vscode/pull/254444#pullrequestreview-3001767448, we can't register an instance listener without removing it once it's disposed